### PR TITLE
Added apidoc for API doc generation and added in-code doc for loading dictionary entries and files

### LINKED ISSUE #12: Create API documentation

### PR DESCRIPTION
**Pull request by hanatgit**
The generated documentation can be seen at https://hanat.work/wp/wp-content/uploads/api/
(We can add a step in auto deployment for documentation update as well later).

Also, during documentation and testing, I made some improvements:
- added mongdb connection caching.
- restructured data structure for loading entries, and refactored how it is shared between load-migrate.ts and loadEntry.ts.

I will complete rest of the documentation and refactoring of the other lambda functions after this PR.

### CODE REVIEW

File: webonary-cloud-api/README.md [hunk 1]
```diff
@@ -34,8 +34,9 @@ WCA is a serverless, cloud-based architecture consisting of:
    1. In the root directory, install required npm packages using `npm install`.
    2. In `lambda` subdirectory, where Lambda functions are stored, install required packages using `npm install`.
 4. Install [AWS command line interface](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) and [configure](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) with your AWS credentials.
-5. Install [AWS CDK command line interface](https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html) by doing `npm install -g aws-cdk`.
+5. GoInstall [AWS CDK command line interface](https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html) by doing `npm install -g aws-cdk`.
```
Comment (hanatgit) [inline webonary-cloud-api/README.md]: GoInstall was fixed to "Install" in subsequent commit

File: webonary-cloud-api/lambda/mongo.ts [hunk 2]
```diff
@@ -5,59 +5,23 @@ let cachedDb: MongoClient;
...
   [key: string]: any;
 }
 
 export async function connectToDB(): Promise<MongoClient> {
-  console.log('=> connect to database');
-
   if (cachedDb) {
-    // console.log('=> using cached database instance');
-    // return Promise.resolve(cachedDb);
+    return Promise.resolve(cachedDb);
```
Comment (hanatgit) [inline webonary-cloud-api/lambda/mongo.ts:9]: I initially had problems with caching, but got it fixed while I was doing refactoring. 
The key was not closing the db connection at the end of Lambda execution in loadEntry.ts

File: webonary-cloud-api/lambda/db.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export interface EntryValue {
+  lang: string;
+  value: string;
+  type?: string;
+}
+
+export interface EntrySense {
+  definitionOrGloss: EntryValue[];
```
Comment (jasonleenaylor) [inline webonary-cloud-api/lambda/db.ts:15]: This will grow, but those changes can come later.

File: webonary-cloud-api/tools/flexXhtmlParser.ts [hunk 4]
```diff
@@ -55,15 +26,20 @@ export class FlexXhtmlParser {
...
       },
     );
   }
 
-  static parseEntry(dictionary: string, entryData: string): Entry {
+  static parseEntry(dictionary: string, entryData: string): LoadEntry {
     const $ = cheerio.load(entryData);
-    const _id = $('div.entry').attr('id') ?? '';
+
+    // NOTE: guid field in Webonary and FLex actually includes the character 'g' at the beginning
```
Comment (jasonleenaylor) [inline webonary-cloud-api/tools/flexXhtmlParser.ts:38]: fyi this is because we used it as an 'id' in xhtml and they must always start with a letter.

**Review by jasonleenaylor — APPROVED** [review]

File: webonary-cloud-api/lambda/db.ts (hunk 3 above)
Comment (hanatgit) [reply webonary-cloud-api/lambda/db.ts:15]: of course...

File: webonary-cloud-api/tools/flexXhtmlParser.ts (hunk 4 above)
Comment (hanatgit) [reply webonary-cloud-api/tools/flexXhtmlParser.ts:38]: ah! makes sense. i've kept the same in wordpress, but i wanted to store in its native format for Mongo.

**Pull request merged**